instrumentation/spring/spring-boot-autoconfigure/src/main/java/io/opentelemetry/instrumentation/spring/autoconfigure/OpenTelemetryAutoConfiguration.java

```diff
@@ -76,143 +54,54 @@ public static class OpenTelemetrySdkConfig {
-          .map(metricExporter -> createPeriodicMetricReader(configProperties, metricExporter))
-          .forEach(meterProviderBuilder::registerMetricReader);
-
-      return meterProviderBuilder.setResource(otelResource).build();
+    public OpenTelemetrySdkComponentLoader openTelemetrySdkComponentLoader(
+        ApplicationContext applicationContext) {
+      return new OpenTelemetrySdkComponentLoader(applicationContext);
     }
 
-    private static PeriodicMetricReader createPeriodicMetricReader(
-        ConfigProperties properties, MetricExporter metricExporter) {
-      PeriodicMetricReaderBuilder metricReaderBuilder =
-          PeriodicMetricReader.builder(metricExporter);
-      Duration interval = properties.getDuration("otel.metric.export.interval");
-      if (interval != null) {
-        metricReaderBuilder.setInterval(interval);
-      }
-      return metricReaderBuilder.build();
+    @Bean
+    public ResourceProvider otelSpringResourceProvider(Optional<BuildProperties> buildProperties) {
+      return new SpringResourceProvider(buildProperties);
     }
 
     @Bean
-    @ConditionalOnMissingBean
-    public Resource otelResource(
-        ConfigProperties config, ObjectProvider<List<ResourceProvider>> resourceProviders) {
-      Resource resource = Resource.getDefault();
-      for (ResourceProvider resourceProvider :
-          resourceProviders.getIfAvailable(Collections::emptyList)) {
-        resource = resource.merge(resourceProvider.createResource(config));
-      }
-      return resource;
+    public ResourceProvider otelDistroVersionResourceProvider() {
+      return new DistroVersionResourceProvider();
     }
 
     @Bean
     // If you change the bean name, also change it in the OpenTelemetryJdbcDriverAutoConfiguration
     // class
     public OpenTelemetry openTelemetry(
-        ObjectProvider<ContextPropagators> propagatorsProvider,
-        SdkTracerProvider tracerProvider,
-        SdkMeterProvider meterProvider,
-        SdkLoggerProvider loggerProvider,
-        ObjectProvider<List<OpenTelemetryInjector>> openTelemetryConsumerProvider) {
-
-      ContextPropagators propagators = propagatorsProvider.getIfAvailable(ContextPropagators::noop);
-
-      OpenTelemetrySdk openTelemetry =
-          OpenTelemetrySdk.builder()
-              .setTracerProvider(tracerProvider)
-              .setMeterProvider(meterProvider)
-              .setLoggerProvider(loggerProvider)
-              .setPropagators(propagators)
-              .build();
-
-      List<OpenTelemetryInjector> openTelemetryInjectors =
-          openTelemetryConsumerProvider.getIfAvailable(() -> Collections.emptyList());
-      openTelemetryInjectors.forEach(consumer -> consumer.accept(openTelemetry));
+        Environment env,
+        OtlpExporterProperties otlpExporterProperties,
+        OtelResourceProperties resourceProperties,
+        PropagationProperties propagationProperties,
+        OpenTelemetrySdkComponentLoader componentLoader,
```

Comment:
Could  `OpenTelemetrySdkComponentLoader` use the `io.opentelemetry.instrumentation.spring.autoconfigure` package and have  a package-private visibilty, or be a nested static class in `OpenTelemetryAutoConfiguration`?